src/coreComponents/common/format/table/unitTests/testTable.cpp

```diff
@@ -230,100 +266,190 @@ TEST( testTable, table2DTable )
+             "|Cras egestas|CoordX| C | CoordZ  | Prev  | Next  |\n"
+             "|            |      |   |         |element|element|\n"
+             "---------------------------------------------------\n"
+             "|           1|     2|3.0|3.0129877|      2|      1|\n"
+             "|           1|     2|3.0|3.0129877|      2|      1|\n"
+             "---------------------------------------------------\n"
              );
 }
 
-TEST( testTable, tableSetMargin )
+TEST( testTable, testCellMerging )
 {
-  ////////////
-  //////// If setMargin used elsewhere make it public and remove comments for this test
-  ////////////
-  //test with tiny margin
-  // {
-  //   TableLayout tableLayout( {
-  //     TableLayout::ColumnParam{{"Colonne 1"}, TableLayout::Alignment::center},
-  //     TableLayout::ColumnParam{{"Colonne 2"}, TableLayout::Alignment::center},
-  //     TableLayout::ColumnParam{{"Colonne 3"}, TableLayout::Alignment::right},
-  //     TableLayout::ColumnParam{{"Colonne 4"}, TableLayout::Alignment::right},
-  //     TableLayout::ColumnParam{{"Prev\nelement"}, TableLayout::Alignment::right},
-  //     TableLayout::ColumnParam{{"Next\nelement"}, TableLayout::Alignment::right},
-  //   }, "InternalWellGenerator well_injector1" );
-
-  //   //tableLayout.setMargin( TableLayout::MarginValue::tiny );
-
-  //   TableData tableData;
-  //   tableData.addRow( "value 1", "long value 1", "3.0034", 3.0129877, "" , 1 );
-  //   tableData.addRow(  "value 1", "long value 2", "100.45", 4.0129877, 1 , 2 );
-
-  //   TableTextFormatter const tableText( tableLayout );
-  //   EXPECT_EQ( tableText.toString( tableData ),
-// "\n------------------------------------------------------------\n"
-// "|           InternalWellGenerator well_injector1           |\n"
-// "------------------------------------------------------------\n"
-// "|Colonne 1| Colonne 2  |Colonne 3|Colonne 4|   Prev|   Next|\n"
-// "|         |            |         |         |element|element|\n"
-// "------------------------------------------------------------\n"
-// "| value 1 |long value 1|   3.0034|3.0129877|       |      1|\n"
-// "| value 1 |long value 2|   100.45|4.0129877|      1|      2|\n"
-// "------------------------------------------------------------\n\n"
-//              );
-// }
+  TableLayout const tableLayout( {
+    TableLayout::Column()
+      .setName( "Cras egestas" ),
+    TableLayout::Column()
+      .setName( "CoordX" ),
+    "C",
+    TableLayout::Column()
+      .setName( "CoordZ" ),
+    TableLayout::Column()
+      .setName( "Prev\nelement" ),
+    TableLayout::Column()
+      .setName( "Next\nelement" )} );
+
+  TableData tableData;
+  tableData.addRow( "ProductA", 1234, 40, "ProductName", 5678, 60 );
+  tableData.addSeparator();
+  tableData.addRow( "P1\nP2\nP3", "2002\n2003\n2004", CellType::MergeNext, 3003, 4004, CellType::MergeNext );
+  tableData.addRow( 3.14f, 2.718f, CellType::MergeNext, 1.618f, 0.577f, CellType::MergeNext );
```

Comment:
Please test with a merged cell which the value is larger than the merged headers.
something like:
```suggestion
  tableData.addRow( 3.14f, 2.718f, CellType::MergeNext, 123123123123123123123, 0.577f, CellType::MergeNext );
```
The expected line is:
```
             "|      ProductA  |    1234  |    40  |   ProductName  |     5678  |       60  |\n"
[...]
             "|          3.14  |   2.718  |  123123123123123123123  |    0.577  |           |\n"
```

Reply:
I think the result of the test should be :
```
  EXPECT_EQ( tableText.toString( tableData ),
             "\n------------------------------------------------------------------------------------------------------------\n"
             "|    Cras egestas     |  CoordX  |     C     |   CoordZ          |     Prev     |           Next           |\n"
             "|                     |          |           |                   |   element    |         element          |\n"
             "------------------------------------------------------------------------------------------------------------\n"
             "|           ProductA  |    1234  |       40  |      ProductName  |        5678  |                      60  |\n"
             "|           ProductA  |      54  |  4564575  |  long size value  |  5454554512  |                      60  |\n"
             "------------------------------------------------------------------------------------------------------------\n"
             "|           ProductA  |      54  |  4564575  |  long size value  |  5454554512  |                      60  |\n"
             "|               3.14  |   2.718  |                        1.618  |       0.577  |                          |\n"
             "|                 P1  |    2002  |       1212121245452145454545  |        4004  |                          |\n"
             "|                 P2  |    2003  |                               |              |                          |\n"
             "|                 P3  |    2004  |                               |              |                          |\n"
             "|  Long product size  |      54  |  4564575  |  long size value  |  5454554512  |                      60  |\n"
             "|     ProductAfdggfd  |    5445  |     4565  |           PrName  |     5454512  |                   64650  |\n"
             "|               3.14  |   2.718  |                        1.618  |       0.577  |                          |\n"
             "------------------------------------------------------------------------------------------------------------\n"
             "|                                                                                                   Item2  |\n"
             "------------------------------------------------------------------------------------------------------------\n"
             "|               1500  |    2500  |                                                                         |\n"
             "------------------------------------------------------------------------------------------------------------\n"
             "|               1.23  |    4.56  |                         7.89  |        0.12  |                      40  |\n"
             "|  Long product size  |      54  |  4564575  |  long size value  |  5454554512  |                      60  |\n"
             "|           ProductA  |      54  |  4564575  |  long size value  |  5454554512  |                      60  |\n"
             "------------------------------------------------------------------------------------------------------------\n"
             "|                 P1  |    2002  |                                                 1212121245452145454545  |\n"
             "------------------------------------------------------------------------------------------------------------\n"
             "|              Alpha  |    1001  |        8  |             Beta  |        2002  |                          |\n"
             "|                     |          |           |            water  |         1.0  |                          |\n"
             "------------------------------------------------------------------------------------------------------------\n"
             );
```